osu.Game/Screens/OnlinePlay/Multiplayer/Match/Playlist/MultiplayerHistoryList.cs

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Collections.Generic;
+using System.Linq;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Game.Online.Rooms;
+using osuTK;
+
+namespace osu.Game.Screens.OnlinePlay.Multiplayer.Match.Playlist
+{
+    /// <summary>
+    /// A historically-ordered list of <see cref="DrawableRoomPlaylistItem"/>s.
+    /// </summary>
+    public class MultiplayerHistoryList : DrawableRoomPlaylist
+    {
+        public MultiplayerHistoryList()
+            : base(false, false, true)
+        {
+        }
+
+        protected override FillFlowContainer<RearrangeableListItem<PlaylistItem>> CreateListFillFlowContainer() => new HistoryFillFlowContainer
```

Comment:
Quite unfortunate that this is required in each implementation. Could be avoided by making an overridable sort function at the `DrawableRoomPlaylist` class level. Would clean things up but not sure it's worth the effort (will give this a try myself).

Reply:
Fundamentally not against it but feels like bundling up too many changes in one PR.

Reply:
Yeah I had a try at this and it wasn't great, so let's go with it as is for now. The `Spacing` being specified in three places is the worst part for me, maybe that can be moved out somehow..